server/admin_management/api/app/routers/router_guards.py

```diff
@@ -168,14 +179,55 @@ async def get_target_user_from_id(
         return target_new_user
 
 
+async def get_request_role_from_id(
+    request: Request,
+    access_control_privilege_service: AccessControlPrivilegeService = Depends(
+        access_control_privilege_service_instance
+    ),
+    role_service: RoleService = Depends(role_service_instance),
+) -> Union[FamRole, None]:
+    """
+    To get role from request
+    For deleting access control privilege method, get the role by access_control_privilege_id
+    For adding access control privilege method, get role through request data
+    """
+    access_control_privilege_id = None
+    if "access_control_privilege_id" in request.path_params:
+        access_control_privilege_id = request.path_params["access_control_privilege_id"]
+
+    if access_control_privilege_id:
+        access_control_privilege = access_control_privilege_service.get_acp_by_id(
+            access_control_privilege_id
+        )
+        return access_control_privilege.role
+    else:
+        try:
+            rbody = await request.json()
+            role_id = rbody.get("role_id")
+            role = role_service.get_role_by_id(role_id)
+            return role  # role could be None.
+
+        # When request does not contains body part.
+        except json.JSONDecodeError:
+            return None
+
+
 async def get_verified_target_user(
     requester: Requester = Depends(get_current_requester),
     target_user: TargetUser = Depends(get_target_user_from_id),
+    role: FamRole = Depends(get_request_role_from_id),
 ) -> TargetUser:
     """
     Validate the target user by calling IDIM web service, and update business Guid for the found BCeID user
     """
-    target_user_validator = TargetUserValidator(requester, target_user)
+    # ignore the role validation here, cause we have other router guards to do that when grant/delete delegated admin
+    # in the case that role is none, that's for granting/deleting application admin with FAM, which has no app_env
+    app_env = role.application.app_environment if role and role.application else None
```

Comment:
we will either get application from role while granting/deleting delegated admin, or set app_env to be None in the case of granting application admin